### PR TITLE
Move all container configuration from build to docker-compose

### PR DESCRIPTION
**Pull request by dcsobral**
This makes testing locally much easier.

### CODE REVIEW

File: key_for_docker [hunk 1]
```diff
@@ -0,0 +1,42 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (alissapajer) [inline key_for_docker:1]: Why are we checking in a private key in plaintext into the repo?

Comment (cryogenian) [reply key_for_docker:1]: Because it's used only for ci test in the repo, right?

Comment (dcsobral) [reply key_for_docker:1]: Right.

**Comment by dcsobral** [comment]
I can't reproduce that error locally. Unfortunately, the "too many authentication failures" error hides the actual authentication error. I don't know what to do next.

**Comment by cryogenian** [comment]
Maybe generate keys instead of using predefined? That's the only reason I can imagine be different from local system and CI

**Comment by dcsobral** [comment]
> Maybe generate keys instead of using predefined? That's the only reason I can imagine be different from local system and CI

It turns out the problem is the owner. On my computer, for whatever reason, the "secret" file end up as root:root, whereas on github it ends up as something else. Maybe it is user the docker vm is running as. I'm going to try switching to docker stack, which supports configuration for that.

File: key_for_docker (hunk 1 above)
Comment (dcsobral) [reply key_for_docker:1]: I'm of two minds here. On one hand, it does trigger a security warning, and it is possible to use it to access the docker vm running in the test server. On the other hand, generating it is one additional step people would have to do to be able to run tests. I'm inclined to add it to the `.gitignore` and mention generating it as a one-time step.
Thoughts?

Comment (cryogenian) [reply key_for_docker:1]: Sounds good

**Comment by dcsobral** [comment]
On a side note, reading https://vsupalov.com/difference-docker-compose-and-docker-stack/ made me think switching to docker swarm is a good idea anyway.

**Pull request merged**